File: backend/app/schemas/rates.py
from pydantic import BaseModel
from datetime import date as date_type


class ExchangeRateItem(BaseModel):
    target_currency: str
    rate: float
    date: date_type
    trend_7d: float | None = None


class ExchangeRateListResponse(BaseModel):
    base: str
    as_of: date_type | None = None
    rates: list[ExchangeRateItem]
    history_7d: dict[str, list[dict[str, str | float]]] | None = None


class InterestRateItem(BaseModel):
    country_code: str
    currency_code: str
    country_name: str
    rate: float
    date: date_type | None = None
    trend_7d: float | None = None


class InterestRateListResponse(BaseModel):
    as_of: date_type | None = None
    rates: list[InterestRateItem]
    history_7d: dict[str, list[dict[str, str | float]]] | None = None

File: frontend/src/types/rates.ts
export interface ExchangeRateItem {
  target_currency: string
  rate: number
  date: string
  trend_7d: number | null
}

export interface ExchangeRateListResponse {
  base: string
  as_of: string | null
  rates: ExchangeRateItem[]
  history_7d: Record<string, Array<{ date: string; rate: string }>> | null
}

export interface InterestRateItem {
  country_code: string
  currency_code: string
  country_name: string
  rate: number
  date: string | null
  trend_7d: number | null
}

export interface InterestRateListResponse {
  as_of: string | null
  rates: InterestRateItem[]
  history_7d: Record<string, Array<{ date: string; rate: string }>> | null
}